library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java

```diff
@@ -444,7 +444,7 @@ private NetworkState checkSizeAllowedForNetwork(int networkType) {
      */
     public boolean isReadyToDownload() {
         synchronized (this) {
-            return isDownloadManagerReadyToDownload() && isClientReadyToDownload();
+            return isClientReadyToDownload() && isDownloadManagerReadyToDownload();
```

Comment:
https://github.com/novoda/download-manager/pull/52 `isDownloadManagerReadyToDownload()` checks if a download has been successfully completed and returns false, therefore `isClientReadyToDownload()` is not invoked. When the other away around the client is asked if they are allowed to download an already downloaded file...